globus_sdk/transfer/client.py

```diff
@@ -334,6 +334,67 @@ def submit_delete(self, data):
         """
         return self.post('/delete', data)
 
+    def make_submit_transfer_data(self, source_endpoint, dest_endpoint,
+                                  transfer_items, label, sync_level=None,
+                                  **kwargs):
```

Comment:
My feeling was that "Label is optional for the API, but let's make it mandatory for the SDK".

@jaswilli Do you really think people will pass the empty string? Or are you more of the camp that the label should be optional, as it is for the API?
I don't feel terribly strongly, but I think any SDK user should be setting the label to identify the program that's starting transfers (because the tasks are still owned by the user, and there's no other way to track that).